src/lib/components/payload-input-with-encoding.svelte

```diff
@@ -22,36 +22,49 @@
   export let id = uuid();
   export let input: string;
   export let encoding: Writable<PayloadInputEncoding>;
+  export let messageType: string;
   export let error = false;
   export let loading = false;
   export let label = translate('workflows.input');
   export let editing = true;
 
-  const clearValues = () => {
-    $encoding = 'json/plain';
-  };
-
-  onDestroy(clearValues);
+  $: {
+    if ($encoding === 'json/plain' && messageType) {
+      messageType = '';
+    }
+  }
 </script>
 
 <div>
   <h5 class="pb-1 text-sm font-medium">{label}</h5>
   <Card class="flex flex-col gap-2">
     <PayloadInput bind:input bind:loading {error} {id} {editing} />
-    <div class="flex items-end {editing ? 'justify-between' : 'justify-end'}">
+    <div
+      class="flex items-end gap-2 {editing ? 'justify-between' : 'justify-end'}"
+    >
       {#if editing}
-        <RadioGroup
-          description={translate('workflows.encoding')}
-          bind:group={encoding}
-          name="encoding"
-        >
-          <RadioInput id="json/plain" value="json/plain" label="json/plain" />
-          <RadioInput
-            id="json/protobuf"
-            value="json/protobuf"
-            label="json/protobuf"
-          />
-        </RadioGroup>
+        <div class="flex w-full flex-col gap-2">
+          <RadioGroup
+            description={translate('workflows.encoding')}
+            bind:group={encoding}
+            name="encoding"
+          >
+            <RadioInput id="json/plain" value="json/plain" label="json/plain" />
+            <RadioInput
+              id="json/protobuf"
+              value="json/protobuf"
+              label="json/protobuf"
+            />
+          </RadioGroup>
+          {#if $encoding === 'json/protobuf'}
+            <Input
+              label={translate('workflows.message-type')}
+              bind:value={messageType}
+              {error}
+              id="messageType"
```

Comment:
Since there isn't a max character limit wondering if we want a `<Textarea />` here instead 🤔

Reply:
Good call out. Generally speaking this is used as a single Proto definition name like `temporal.api.v1.ThingWithType` so it shouldn't need a textarea